packages/office-ui-fabric-react/src/components/Modal/Modal.Props.ts

```diff
@@ -66,4 +69,9 @@ export interface IModalProps extends React.Props<Modal>, IWithResponsiveModeStat
    * ARIA id for the subtitle of the Modal, if any
    */
   subtitleAriaId?: string;
-}
\ No newline at end of file
+
+  /**
+  * Callback to dynamically set the selector for the first focusable within the modal
+  */
+  getFirstFocusableSelector?: () => string;
```

Comment:
Minor suggestion on this; maybe a slight tweak to the name. Also it could take either a function or a string.

```tsx
initialFocusSelector: string | () => string;
```

Most cases the user will have some static selector string to use and won't need a function. 

It would also be nice to have this same property on FocusZone for symmetry.

I thought that `defaultActiveElement` was this, but it seems to be both initializing the active element and setting focus to that element on mount, which seems wrong to me (I don't think I added that property but my name is on the blame history, maybe I did long ago?) I think we should consider deprecating the `defaultActiveElement` property because it's confusing. I could see the potential for a `setInitialFocus` prop to pull focus on mounting, to replace the "focus on mount" behavior.

Reply:
Good point on the selector prop type. This is mutually exclusive with the string selector, so is makes sense to use the same prop.

In general, the differing focus behavior between FocusZone and FocusTrapZone is a bit surprising. I hadn't looked closely before, and was sort of expecting them to be the same. It sounds like adjusting the semantics of FocusZone will require some further thoughts and is slightly outside the scope of this PR. :)